Add detail in docs for histogram indicator type (#163905)
elastic/kibana: x-pack/plugins/observability/dev_docs/slo.md

- Custom KQL
- Custom Metric
- Custom Histogram

For the APM SLIs, customer can provide the service, environment, transaction name and type to configure them. For the **APM Latency** SLI, a threshold in milliseconds needs to be provided to discriminate the good and bad responses (events). For the **APM Availability** SLI, we use the `event.outcome` as a way to discriminate the good and the bad responses(events). The API supports an optional kql filter to further filter the apm data.

The **custom KQL** SLI requires an index pattern, an optional filter query, a numerator query, and denominator query. A custom `timestampField` can be provided to override the default @timestamp field.

The **custom Metric** SLI requires an index pattern, an optional filter query, a set of metrics for the numerator, and a set of metrics for the denominator. A custom `timestampField` can be provided to override the default @timestamp field.

The **custom Histogram** SLI requires an index pattern, an optional filter query, and an optional `timestampField`. `good` represents the numerator and `total` represents the denominator, and both require the following fields:

> * field - the field used to aggregate good/total events.

Reviewer: ```suggestion
* field - the histogram field used to aggregate good/total events.
```

}'
```

</details>

### Custom Histogram

<details>

> <summary>95.0% of transactions with latency < 300 over the last 7 days</summary>

Reviewer: ```suggestion
<summary>95.0% of transactions with latency between 0 and 300ms over the last 7 days</summary>
```

```
curl --request POST \
  --url http://localhost:5601/cyp/api/observability/slos \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'Content-Type: application/json' \
  --header 'kbn-xsrf: oui' \
  --data '{
    "name": "My SLO Name",
    "description": "My SLO Description",
    "indicator": {
        "type": "sli.histogram.custom",
        "params": {
            "filter": "",

>             "index": "transactions,

Reviewer: ```suggestion
            "index": "transactions-*",
```

            "timestampField": "custom_timestamp",

>             "good": {
>                 "aggregation": "value_count",
>                 "field": "txn",
>                 "filter": "latency < 300"
>             },

Reviewer: ```suggestion
            "good": {
                "aggregation": "range",
                "field": "latency",
                "from": 0,
                "to": 300
            },
```

>             "total": {
>                 "aggregation": "value_count",
>                 "field": "txn"
>             }

Reviewer: ```suggestion
            "total": {
                "aggregation": "value_count",
                "field": "latency"
            }
```
